pkg/dns/treecache.go

```diff
@@ -130,32 +90,38 @@ func (cache *TreeCache) GetSubCache(path ...string) *TreeCache {
 	return childCache
 }
 
-func (cache *TreeCache) SetSubCache(key string, subCache *TreeCache, path ...string) {
-	cache.m.Lock()
-	defer cache.m.Unlock()
+func (cache *TreeCache) setSubCache(key string, subCache *TreeCache, path ...string) {
```

Comment:
yes, thats fine. It means I need to have a decorator struct which wraps the non-thread-safe functions on the cache strut.